frontend/src/metabase/dashboard/components/DashCard/DashCardParameterMapper/DashCardCardParameterMapper.jsx

```diff
@@ -113,6 +111,33 @@ export function DashCardCardParameterMapper({
   const virtualCardType = getVirtualCardType(dashcard);
   const isNative = isNativeDashCard(dashcard);
 
+  let selectedMappingOption;
+  if (isVirtual || isAction || isNative) {
+    selectedMappingOption = target
+      ? mappingOptions.find(option =>
+          _.isEqual(normalize(option.target), normalize(target)),
+        )
+      : undefined;
+  } else {
+    const stageIndex = -1;
+    const columns = Lib.visibleColumns(question.query(), stageIndex);
+
+    selectedMappingOption = target
+      ? mappingOptions.find(mappingOption => {
+          const [index1, index2] = target
+            ? Lib.findColumnIndexesFromLegacyRefs(
+                question.query(),
+                stageIndex,
+                columns,
+                [normalize(target[1]), normalize(mappingOption.target[1])],
+              )
+            : [-1];
+
+          return index1 >= 0 && index1 === index2;
+        })
+      : undefined;
```

Comment:
There's a lot of stuff going on here, and it could be hard to read, IMO. We're checking if `target` is null-ish on line 127, but we already checked that on line 125. We're also doing a similar check for `isVirtual || isAction || isNative` block above; maybe we could do an early return and simplify both of these blocks?

Reply:
simplified function and moved it to utils [480e5e3](https://github.com/metabase/metabase/pull/38545/commits/480e5e3220108fac3861e2ae2f47a2319f9c9839)